beacon-chain/core/altair/sync_committee.go

```diff
@@ -221,3 +242,85 @@ func SubnetsForSyncCommittee(state iface.BeaconStateAltair, i types.ValidatorInd
+//    if compute_sync_committee_period(get_current_epoch(state)) == compute_sync_committee_period(next_slot_epoch):
+//        sync_committee = state.current_sync_committee
+//    else:
+//        sync_committee = state.next_sync_committee
+//
+//    # Return pubkeys for the subcommittee index
+//    sync_subcommittee_size = SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT
+//    i = subcommittee_index * sync_subcommittee_size
+//    return sync_committee.pubkeys[i:i + sync_subcommittee_size]
+func SyncSubCommitteePubkeys(st iface.BeaconStateAltair, subComIdx types.CommitteeIndex) ([][]byte, error) {
+	nextSlotEpoch := helpers.SlotToEpoch(st.Slot() + 1)
+	currEpoch := helpers.SlotToEpoch(st.Slot())
+
+	var syncCommittee *pb.SyncCommittee
+	var err error
+	if SyncCommitteePeriod(currEpoch) == SyncCommitteePeriod(nextSlotEpoch) {
+		syncCommittee, err = st.CurrentSyncCommittee()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		syncCommittee, err = st.NextSyncCommittee()
+		if err != nil {
+			return nil, err
+		}
+	}
+	subCommSize := params.BeaconConfig().SyncCommitteeSize / params.BeaconConfig().SyncCommitteeSubnetCount
+	i := uint64(subComIdx) + subCommSize
+	endOfSubCom := i + subCommSize
+	pubkeyLen := uint64(len(syncCommittee.Pubkeys))
+	if endOfSubCom > pubkeyLen {
+		return nil, errors.Errorf("end index is larger than array length: %d > %d", endOfSubCom, pubkeyLen)
+	}
+	return syncCommittee.Pubkeys[i:endOfSubCom], nil
+}
+
+// IsSyncCommitteeAggregator checks whether the provided signature is for a valid
+// aggregator.
+// def is_sync_committee_aggregator(signature: BLSSignature) -> bool:
+//    modulo = max(1, SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT // TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE)
+//    return bytes_to_uint64(hash(signature)[0:8]) % modulo == 0
+func IsSyncCommitteeAggregator(sig []byte) bool {
+	modulo := mathutil.Max(1, params.BeaconConfig().SyncCommitteeSize/params.BeaconConfig().SyncCommitteeSubnetCount/params.BeaconConfig().TargetAggregatorsPerSyncSubcommittee)
+	hashedSig := hashutil.Hash(sig)
+	return bytesutil.BytesToUint64BigEndian(hashedSig[:8])%modulo == 0
+}
+
+// SyncCommitteeSigningRoot returns the signing root from the relevant provided data.
+//
+// def get_sync_committee_selection_proof(state: BeaconState,
+//                                       slot: Slot,
+//                                       subcommittee_index: uint64,
+//                                       privkey: int) -> BLSSignature:
+//    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF, compute_epoch_at_slot(slot))
+//    signing_data = SyncAggregatorSelectionData(
+//        slot=slot,
+//        subcommittee_index=subcommittee_index,
+//    )
+//    signing_root = compute_signing_root(signing_data, domain)
+//    return bls.Sign(privkey, signing_root)
+func SyncCommitteeSigningRoot(st iface.BeaconState, slot types.Slot, comIdx types.CommitteeIndex) ([32]byte, error) {
```

Comment:
Better name: SyncSelectionProofSigningRoot

`SyncCommitteeSigningRoot` can be referred to multiple things here